ports/raspberrypi/common-hal/busio/SPI.c

```diff
@@ -263,8 +263,8 @@ bool common_hal_busio_spi_write(busio_spi_obj_t *self,
 
 bool common_hal_busio_spi_read(busio_spi_obj_t *self,
     uint8_t *data, size_t len, uint8_t write_value) {
-    uint32_t data_out = write_value << 24 | write_value << 16 | write_value << 8 | write_value;
-    return _transfer(self, (const uint8_t *)&data_out, MIN(4, len), data, len);
```

Comment:
Why doesn't this work? It'd be good to avoid the memset if we can. (It should just DMA the same values over and over.)

Reply:
Maybe I misunderstand what it did. It repeats the output as many times as needed for the input?  And the total is the max of the two? or the input size?

Reply:
Correct, the transfer code won't increment the source address if the overall length is longer than the output length: https://github.com/adafruit/circuitpython/blob/main/ports/raspberrypi/common-hal/busio/SPI.c#L192

Reply:
thanks, I learned something!